### PR TITLE
Fix beat divisor control selecting invalid divisors on drag end

### LINKED ISSUE #8326: Beat Divisor Control value can be set outside of valid divisors
**Describe the bug:** When dragging marker it is possible to set value and position of marker outside of valid divisors, it happens only if you start dragging > marker jump > release mouse > it will land on wherever your mouse position was. 
That leading to having all red and small divisors in beatmap editor when it set to 1/7, 1/11 etc.

**Screenshots or videos showing encountered issue:** 
https://imgur.com/a/3nm2lSl

**osu!lazer version:** 2020.317.0

I don't know how to submit a fix.
So there is a fix, in BeatDivisorControl.cs TickSliderBar class, we need to override OnDragEnd to prevent base class from changing value of the slider.
`
protected override void OnDragEnd(DragEndEvent e) {}
`

### PR DESCRIPTION
**Pull request by Fuewburvpoa**
Resolves #8326.

### CODE REVIEW

File: osu.Game/Screens/Edit/Compose/Components/BeatDivisorControl.cs [hunk 1]
```diff
@@ -279,6 +279,10 @@ protected override void OnDrag(DragEvent e)
                 handleMouseInput(e.ScreenSpaceMousePosition);
             }
 
+            protected override void OnDragEnd(DragEndEvent e)
+            {
+            }
```
Comment (bdach) [inline osu.Game/Screens/Edit/Compose/Components/BeatDivisorControl.cs]: While the diagnosis that the problem lies with the base implementation of `OnDragEnd` seems correct, I'm not overly convinced that silencing the event outright is a better solution than just mirroring `OnDrag`:

```suggestion
            protected override void OnDragEnd(DragEndEvent e)
            {
                handleMouseInput(e.ScreenSpaceMousePosition);
            }
```

**Review by bdach — COMMENTED** [review]
Could be nice to have a test for this to prevent regressions, although test coverage in editor is currently not that high overall.

**Comment by Fuewburvpoa** [comment]
I will try to write test for this case tomorrow.

**Comment by Fuewburvpoa** [comment]
Added 2 tests.
TestBindableBeatDivisor - test BindableBeatDivisor Next() and Previous() methods.
TestMouseInput - test for #8326

I have one question, is it fine to get location of private elements by simply getting position of the object and finding offset manually? For example in this case i wanted to get position of the Marker that is private member inside private class, so instead of messing up with reflection i just got coords manually.

**Comment by bdach** [comment]
@Fuewburvpoa Your attempt was pretty close! I've just pushed in an adjusted version of the test, that looks for the appropriate drawables in the hierarchy and uses their coordinates to perform the mouse movements.

The selection of the items is a little bit hacky (i.e. moving the marker by getting its inner `EquilateralTriangle` relies on implementation details of the control) but I opted not to expose any of the private inner classes in `BeatDivisorControl`.

**Comment by Fuewburvpoa** [comment]
> @Fuewburvpoa Your attempt was pretty close! I've just pushed in an adjusted version of the test, that looks for the appropriate drawables in the hierarchy and uses their coordinates to perform the mouse movements.
> 
> The selection of the items is a little bit hacky (i.e. moving the marker by getting its inner `EquilateralTriangle` relies on implementation details of the control) but I opted not to expose any of the private inner classes in `BeatDivisorControl`.

I see, thank you for good example, will use it in future cases.

**Comment by peppy** [comment]
@Fuewburvpoa can you explain why adding the `OnDragEnd` is fixing this? Is it due to bypassing the `base.` invocation? Always helps the review process to mention this in the PR description or commit.

**Comment by bdach** [comment]
Yup, the base call in `OnDragEnd` does the normal slider bar logic of clamping position to the nearest tick which breaks the beat divisor as not all tick values are allowed in this case.

**Review by peppy — APPROVED** [review]

**Pull request merged**